netscaler/netscaler_resource.go

```diff
@@ -123,16 +156,37 @@ func (c *NitroClient) doHTTPRequest(method string, url string, bytes *bytes.Buff
 	if err != nil {
 		return []byte{}, err
 	}
+	// For login request 'sessionid' is present in cookies
+	for _, cookie := range resp.Cookies() {
+		if cookie.Name == "sessionid" {
+			sessionid, err := url.QueryUnescape(cookie.Value)
+			if err == nil {
+				c.sessionid = sessionid
+			}
+		}
+	}
 	log.Println("[DEBUG] go-nitro: response Status:", resp.Status)
-	return respHandler(resp)
+	body, err := respHandler(resp)
```

Comment:
One of the intention for auto clearing of session-id (in case of timeout error) is so that user need not worry about checking the timeout error. If user doesn't check for timeout error and timeout happens, all subsequent requests will start failing. With sesion-id auto clearing logic, one request will fail and all other subsequent requests will use basic auth as fall back if re-login is not done (can be done by checking if session-id is nil).

Currently public functions for operations don't return response-body (AddResource, UpdateResource, etc) and changing them may cause API inconsistency. And user will need response-body for checking the error-code.

I added clearing of session-id in case of timeout inside function `doHTTPRequest` so that logic is common for all the operations(ADD/UPDATE/DELETE).

Reply:
if it is auto-cleared then should we re-login automatically? (probably not). Also occurred to me that there should be a way to automatically back-off if there is authentication failure. Otherwise, automation built on top of this client might spam the NS if someone changes / rotates the password without the client knowing.

Reply:
Relogin support within the library can be supported with maximum retries configured. But that can be supported later maybe and not be part of this PR. User has to take care of re -logging in currently in case of timeout error or fallback to basic-auth.